### DISCUSSION TITLE
2.0 Starting Points

### DISCUSSION THREAD

**Discussion by caseybaggz** [pluralsight/design-system#2013] [Ideas]
## DOCS
- Use Webpack/Jest design which includes history of API versions

## STACK
- Yarn 2 / PnP???
- TS
- Lib agnostic???

## DESIGN
- single lib???
- headless??? (what do we want to be responsible for? Only a11y & styles?)
- decoupled group-based packages (i.e. core, flow, skills, acg, actions, form-actions, etc.)???
- setup function which will add core, normalize stuff vs installing new package???

## RELEASE
- research Material UI, Chakra, Ant, React, Jest, Svelte
- use common release approach of stable, next, main (dev)
- milestones = version/feature determine, not timeline

## STRAT
- send out survey / Sprint
- fix bugs?
- tackle tech debt while creating new project for next phase of lib?

**Comment by caseybaggz** [comment]
MUI has some great documentation on their version & release strategy - might not be a bad idea to include this in our docs as well:
https://mui.com/versions/#versioning-strategy

**Reply by caseybaggz** [reply]
TL;DR

#### Branch strategy
- `main`, stable
- `next`, future rc
#### Release cycle
- no set dates
- plan on at least 1 major a year and 3 minors within a major
- patch can be on a weekly basis *if needed* (hotfixes)

**Reply by caseybaggz** [reply]
Jest team has no committed release cycle - anything from 2 major a month to 1 major a year.

**Reply by caseybaggz** [reply]
Ant team

#### Release Schedule
- Weekly release: patch version at the end of every week for routine bugfix (anytime for urgent bugfix).
- Monthly release: minor version at the end of every month for new features.
- Major version release is not included in this schedule for breaking change and new features.

**Comment by caseybaggz** [comment]
CI tools to help enhance checks and stability:

- *Codesandbox* for API/integration testing
- *GH actions* to help speed up the overall process (tagging, release, etc.)
- *Codecov* (any testing accountability reporter)
- *Saucelabs* for visual-ui check (can we automate comparing Figma design file to what is being shipped?)
- *Sonarlint* for overall code quality checks
- *Danger* for checking PR size vs release number which helps encourage cleaner releases and better engineering accountability
- *Vercel* for PR previews of the docs to help streamline reviews

**Reply by caseybaggz** [reply]
Jest team uses CircleCI to automate node testing, GH Actions for issue management, and manually releases with whichever admin.

**Comment by caseybaggz** [comment]
Better labels for automation (i.e., docs, PR: outdated, bug, etc.)

**Comment by caseybaggz** [comment]
Meet different OS platform tooling
